fast_llm/data/data/gpt/data.py

```diff
@@ -23,6 +26,20 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclasses.dataclass
+class GPTBatch:
+    token_ids: torch.Tensor
+    loss_masking_spans: list[torch.Tensor] | None
+
+
+def gpt_data_collate_fn(batch: list[GPTSample]) -> GPTBatch:
+    stacked_ids = np.stack([sample.token_ids for sample in batch])
+    stacked_spans = None
+    if batch[0].loss_masking_spans is not None:
+        stacked_spans = [torch.from_numpy(sample.loss_masking_spans) for sample in batch]
+    return GPTBatch(token_ids=torch.from_numpy(stacked_ids), loss_masking_spans=stacked_spans)
```

Comment:
```suggestion
def gpt_data_collate_fn(batch: list[GPTSample], use_loss_masking_spans: bool) -> GPTBatch:
    stacked_ids = np.stack([sample.token_ids for sample in batch])
    stacked_spans = None
    if use_loss_masking_spans:
        stacked_spans = [torch.from_numpy(sample.loss_masking_spans) for sample in batch]
    return GPTBatch(token_ids=torch.from_numpy(stacked_ids), loss_masking_spans=None)
```
is that possible?
what does `loss_masking_spans = None` mean? usage of loss masking spans turned off globally?

Reply:
yes, that was the hacky way of figuring it out. Your suggestion is much cleaner